package/gluon-ebtables-split-igmpmld-domain/files/lib/gluon/ebtables/105-mcast-drop-mld

```diff
@@ -0,0 +1,9 @@
+rule('ICMPV6_OUT -p IPv6 --ip6-protocol ipv6-icmp --ip6-icmp-type 130 -j DROP') -- MLD Query
+rule('ICMPV6_OUT -p IPv6 --ip6-protocol ipv6-icmp --ip6-icmp-type 131 -j DROP') -- MLDv1 Report
+rule('ICMPV6_OUT -p IPv6 --ip6-protocol ipv6-icmp --ip6-icmp-type 132 -j DROP') -- MLDv1 Done
+rule('ICMPV6_OUT -p IPv6 --ip6-protocol ipv6-icmp --ip6-icmp-type 143 -j DROP') -- MLDv2 Report
+
+rule('ICMPV6_IN -p IPv6 --ip6-protocol ipv6-icmp --ip6-icmp-type 130 -j DROP', 'nat') -- MLD Query
+rule('ICMPV6_IN -p IPv6 --ip6-protocol ipv6-icmp --ip6-icmp-type 131 -j DROP', 'nat') -- MLDv1 Report
+rule('ICMPV6_IN -p IPv6 --ip6-protocol ipv6-icmp --ip6-icmp-type 132 -j DROP', 'nat') -- MLDv1 Done
+rule('ICMPV6_IN -p IPv6 --ip6-protocol ipv6-icmp --ip6-icmp-type 143 -j DROP', 'nat') -- MLDv2 Report
```

Comment:
Of course, that could be done too. I think I had even started like that but found the "double negation" a bit confusing to read then if everything is in MULTICAST_IN/OUT directly.

Like, everything is allowed by default, then multicast is dissallowed by default if that package was added, but not everything is dissallowed, ICMPV6 is exempt. But hey, then again, there are some rules a little earlier in the same MULTICAST_IN/OUT chains, again dropping some other ICMPV6 types. And I was then already starting to loose myself with all that :).

So the idea was to instead make the rules more readable by saying right from the start that some ICMPV6 types might be allowed and people can have a closer look within the according ICMPV6 chains then, separating all non-ICMPV6 rules from any concerns with ICMPV6.